### DISCUSSION TITLE
Datum vergleichen und Value ändern

### DISCUSSION THREAD

**Discussion by alicie232** [neuefische/questions#27] [Web] [answered]
Hallo!

Ich möchte, dass meine Checkboxen am nächsten Tag immer _false_ sind.
(alle Aufgaben, egal ob erledigt (abgehakt) oder nicht, sollen am nächsten Tag wieder als unerledigt angezeigt werden, [siehe User Story 3](https://github.com/alicie232/capstone-project/issues/3))

Aktuell speichere ich das Datum im localStorage (checkedAt: "27.7.2022") mit folgender Funktion:

```
let date = new Date();

const dailyTasks = [
  {task: 'Räume lüften', isChecked: false, checkedAt: date.toLocaleDateString(), id: nanoid()},
...
```
[Hier gehts zum entsprechenden Commit](https://github.com/alicie232/capstone-project/commit/fca1a329737621b21e664f6a55ce0da30ac0523b)

Macht das so schonmal Sinn? 

Mir fehlt der richtige Gedanke, wie ich weiter mache. Vergleiche ich das Datum aus dem local Storage, ob es kleiner ist als das aktuelle Datum? Oder muss ich da ganz anders ran gehen?
Wenn ja, freue ich mich über einen Anhaltspunkt, wie man das am besten umsetzt :) oder einen besseren Vorschlag.

[Hier gehts zu meiner App, mit local Storage, aber noch ohne Date Function](https://capstone-project-nu-five.vercel.app/)

Vielen Dank!

**Comment by actyralf** [comment, marked as answer]
Ich denke, die Idee ist richtig. Unmittelbar nach dem Laden der Daten aus dem localStorage musst du dann einmal über alle Tasks iterieren und Einträge mit "veraltetem" `checkedAt` auf `false` setzen. Erst dann dieses geänderte Array in den State setzen (und auch im localStorage aktualisieren).
